addon/-task-instance.js

```diff
@@ -235,7 +253,7 @@ let taskInstanceAttrs = {
    * @instance
    */
   cancel() {
-    if (this.isCanceled || this.isFinished) { return; }
+    if (this.get('isFinished')) { return; }
```

Comment:
It wasn't possible for isCanceled to be set without isFinished being set, because the only code path that changed `isCanceled` also set `isFinished` to true.